VocaDbWeb/Scripts/Shared/WebLinkMatcher.ts

```diff
@@ -944,7 +944,10 @@ export class WebLinkMatcher {
 			cat: WebLinkCategory.Official,
 		},
 		{
-			url: 'nicovideo.jp/mylist/',
+			// matching both
+			// https://www.nicovideo.jp/mylist/000 (old URL style) and
+   			// https://www.nicovideo.jp/user/000/mylist/000 (new URL style)
+			url: '/mylist/',
```

Comment:
I would prefer to have two entries in `WebLinkMatcher.ts` for the new and the old style rather than including this one matcher without a host

Reply:
I initially did that, but then I realized that the one with the host would be redundant as the short one would match a superset; but I also understand the argument that it's stylistically better

Reply:
I'm really sorry for the late reply. My main argument for two separate entries is to still keep the host (https://www.nicovideo.jp) in the matcher to keep URLs who just happen to include `/mylist/` from matching with this matcher

Reply:
My understanding is that
- `https://www.nicovideo.jp/mylist/000` can be matched with `nicovideo.jp/mylist/` (current behavior), but
- `https://www.nicovideo.jp/user/000/mylist/000` must be matched with the host-less `/mylist/` (proposed edit), and cannot have a host (ideally `nicovideo\.jp/user/\d+/mylist/` if represented in regex) due to the simplistic use of `indexOf` (not regex) that cannot handle the necessary wildcard for the intervening `/user/000/`, and
- `nicovideo.jp/mylist/` would then be a redundant test to `/mylist/`.

This \*does\* allow "false positives" for other hosts, but I am not sure if this would happen in practice.